pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -638,6 +638,9 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 	}
 	if gracePeriodOverride != nil {
 		gracePeriod = *gracePeriodOverride
+		if gracePeriod < 0 {
+			gracePeriod = 0
```

Comment:
In this case the gracePeriod is an error. It makes sense to me to default this back to the original value of gracePeriod instead of setting it to zero (which is no grace period).

Reply:
There are a few more placed within the PR where the gracePeriodOveride should be ignored - on error - and the gracePeriod should be preserved.

Reply:
This is to deal with existing negative values, which have a meaning more like 0.

https://github.com/kubernetes/kubernetes/pull/98507#issuecomment-769327287
https://github.com/kubernetes/kubernetes/issues/98506#issuecomment-774598217
@liggitt recommended is clip to 0